src/Raven.Server/Utils/TypeConverter.cs

```diff
@@ -142,6 +144,47 @@ private static object ToBlittableSupportedType(object root, object value, bool f
             {
                 NestingLevelTooDeep(root);
             }
+             if (value is JsValue js)
+            {
+                if (js.IsNull() || js.IsUndefined())
+                    return null;
+                if (js.IsString())
+                    return js.AsString();
+                if (js.IsBoolean())
+                    return js.AsBoolean().ToString(); // avoid boxing the boolean
+                if (js.IsNumber())
+                    return js.AsNumber();
+                if (js.IsDate())
+                    return js.AsDate().ToDateTime();
+                //object wrapper is an object so it must come before the object
+                if (js is ObjectWrapper ow)
+                {
+                    var target = ow.Target;
+                    switch (target)
+                    {
+                        case LazyStringValue lsv:
+                            return lsv;
+                        case LazyCompressedStringValue lcsv:
+                            return lcsv;
+                        case LazyNumberValue lnv:
+                            return lnv; //should be already blittable supported type.
+                    }
+                    ThrowInvalidObject(js);
+                }
+                //Array is an object in Jint
+                else if (js.IsArray())
+                {
+                    var arr = js.AsArray();
+                    return EnumerateArray(arr);
```

Comment:
You need to take into account the value of 'flattenArrays' and return DynamicJsonArray. Check how we do it with non-js objects.

Reply:
Ok, we should do that